File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    queue = []
    for i in range(len(s)):
        if s[i] == '(':
            queue.append('(')
        else:
            if not queue or queue.pop() !='(':
                return False
    
    if len(queue) != 0:
        return False
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let check = 0 
    for(const ob1 of s){
        if (ob1 == '('){
            check ++
        }
        else{
            if(check){
                check --;
            }else{
                return false
            }
        }
    }
    if (!check){
        return true
    } else{
        return false
    }
}